bin/presubmit.dart

```diff
@@ -0,0 +1,24 @@
+import 'dart:async';
+import 'dart:io';
+import 'package:io/io.dart';
+
+Future<Null> main() async {
+  final ProcessManager manager = new ProcessManager();
+  Process spawn;
+
+  print('Running test...');
+  spawn = await manager.spawn('pub', ['run', 'test']);
+  await spawn.exitCode;
+
+  print('Formatting config.json...');
+  spawn = await manager.spawn('bin/configlet', ['fmt', '.']);
```

Comment:
I'm not so sure about omitting configlet lint. Because there could be a human error. Maybe run the configlet lint _after_ formating to make sure it is infact a human error.

Reply:
Also make sure configlet actually exists before running it. Otherwise fetch it.

Reply:
I'm not sure I understand. The command `bin/configlet lint .` is in the `check_formatting.dart` script. I separated the commands into two scripts as @Stargator suggested in #61.

Reply:
Oh, I didn't realize check formating was running before this script. My mistake 🙈. Although do we need to fetch the configlet twice? I'm not sure if the `before script` and `script` sections of Travis configuration  are isolated.

Reply:
That's ok. Well, `check_formatting.dart` and `presubmit.dart` are independent, so you can run them in whichever order you like, though I think it makes more sense to run `check_formatting.dart` first.

No, you only need to fetch the configlet once, but that's a good point. Since you're required to have a configlet to run these scripts, perhaps they could fetch it if one doesn't exist already.

I'm not sure about the Travis configuration. I believe they're isolated, but maybe @Stargator and @kytrinyx can enlighten us.

Reply:
Regardless of Travis configuration being isolated, both of these scripts will still need to check if configlet exists, and fetch it if not. Which is not that much code duplication but I think we can improve on this.

Reply:
I totally agree.